File: c_sharp/src/org/ldk/structs/LSPS0ServiceHandler.cs
using org.ldk.impl;
using org.ldk.enums;
using org.ldk.util;
using System;

namespace org { namespace ldk { namespace structs {


/**
 * The main server-side object allowing to send and receive bLIP-50 / LSPS0 messages.
 */
public class LSPS0ServiceHandler : CommonBase {
	internal LSPS0ServiceHandler(object _dummy, long ptr) : base(ptr) { }
	~LSPS0ServiceHandler() {
		if (ptr != 0) { bindings.LSPS0ServiceHandler_free(ptr); }
	}

}
} } }

File: src/main/java/org/ldk/structs/LSPS0ServiceHandler.java
package org.ldk.structs;

import org.ldk.impl.bindings;
import org.ldk.enums.*;
import org.ldk.util.*;
import java.util.Arrays;
import java.lang.ref.Reference;
import javax.annotation.Nullable;


/**
 * The main server-side object allowing to send and receive bLIP-50 / LSPS0 messages.
 */
@SuppressWarnings("unchecked") // We correctly assign various generic arrays
public class LSPS0ServiceHandler extends CommonBase {
	LSPS0ServiceHandler(Object _dummy, long ptr) { super(ptr); }
	@Override @SuppressWarnings("deprecation")
	protected void finalize() throws Throwable {
		super.finalize();
		if (ptr != 0) { bindings.LSPS0ServiceHandler_free(ptr); }
	}

}
